src/main/java/org/mastodon/mamut/launcher/RecentProjectsPanel.java

```diff
@@ -82,13 +89,20 @@ private void remakeGUI( final Consumer< String > projectOpener )
 
 		if ( recentProjects != null && !recentProjects.isempty() )
 		{
+			gbc.gridy++;
+			final JLabel lblTitleHint = new JLabel( "Double-click to open the containing folder." );
+			lblTitleHint.setFont( lblTitleHint.getFont().deriveFont( lblTitleHint.getFont().getStyle() | Font.ITALIC ) );
+			lblTitleHint.setHorizontalAlignment( SwingConstants.CENTER );
+			add( lblTitleHint, gbc );
+
 			for ( final String projectPath : recentProjects )
 			{
 				gbc.gridy++;
 				gbc.gridx = 0;
 				final JTextArea ta = new JTextArea( projectPath );
 				ta.setEditable( false );
 				ta.setLineWrap( true );
+				ta.addMouseListener( new MouseDblClickOpenPath( projectPath ) );
```

Comment:
action no. 1: the mouse click on the text field is activated here, here it all starts

it creates a dedicated mouse listener for exactly this particular text field item -> see action no. 2

it passes further the string with the text as it is in the text field
the string can (theoretically) be

- a path to a regular file `.mastodon`
- a path to a regular folder